.github/workflows/cluster-integration-tests-kind.yaml

```diff
@@ -0,0 +1,41 @@
+name: Cluster-based Integration Tests (KIND)
+on: [push, pull_request]
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          token: ${{ github.token }}
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+      - name: Install Argo
+        run: |
+          # Download the binary
+          curl -sLO https://github.com/argoproj/argo-workflows/releases/download/v3.1.1/argo-linux-amd64.gz
+          # Unzip
+          gunzip argo-linux-amd64.gz
+          # Make binary executable
+          chmod +x argo-linux-amd64
+          ./argo-linux-amd64 version
+      - name: Create Kind Cluster
+        uses: helm/kind-action@v1.1.0
+        with: 
+          version: v0.11.1
+          node_image: kindest/node:v1.21.1
+          cluster_name: adelphi-integration-tests
+          config: helm/adelphi/tests/kind-config-3-workers.yaml
+      - name: Install Chart and Run Workflow
+        run: helm install adelphi helm/adelphi -n cass-operator -f helm/adelphi/tests/kind-values-trunk.yaml --debug --wait || true
+      - name: Install Chart and Run Workflow (2nd attempt in case of timeout)
+        run: helm install adelphi helm/adelphi -n cass-operator -f helm/adelphi/tests/kind-values-trunk.yaml --debug || true
+      - name: Wait for Workflow Server ready status
+        run: |
+          sleep 30
+          kubectl get all -A
+          ./argo-linux-amd64 get -n cass-operator @latest
+          kubectl wait pod --for=condition=Ready -l app=argo-server -n cass-operator --timeout=10m
+      - name: Argo Wait for Completion
+        run: |
+          ./argo-linux-amd64 watch -n cass-operator @latest
```

Comment:
`argo watch` is not ideal here because it will spam the logs, but I wouldn't like to hold this back because of it, we can improve it later.
Ideally, we should propose/contribute a refresh rate argument for the Argo project.

Reply:
One one hand, it might be verbose, but on the other we've seen many times in K8ssandra how difficult it can be to debug issues that we see in the GHA workflows, so having the logging will be quite useful when things go wrong.  As long as it's not so much that we hit log size limits or anything like that, I don't think it's a huge problem and in some ways desirable.

Reply:
Alright, sounds good. So far we haven't hit any limits and it actually came in handy a couple of times while debugging during development.